lib/report_github_statuses/runner.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require_relative './buildkite'
+require_relative './github/client'
+
+require 'yaml'
+
+module ReportGithubStatuses
+  class Runner
+    class << self
+      def run(step_status_config_path: nil)
+        step_status_config_path ||= ENV['BUILDKITE_PLUGIN_REPORT_GITHUB_STATUSES_STEP_STATUS_CONFIG_PATH']
+
+        new(YAML.load_file(step_status_config_path)).run
+      end
+    end
+
+    attr_reader :step_status_config
+
+    def initialize(step_status_config)
+      @step_status_config = step_status_config
+    end
+
+    def run
+      passed_jobs, failed_and_skipped_jobs = all_jobs_for_build.partition do |job|
+        job.dig("node", "state") == "SKIPPED" || job.dig("node", "passed")
+      end
```

Comment:
These variable names don't seem like they're matching the partitioning being done. Looks like we're separating into `passed_and_skipped_jobs` and `failed_jobs`.